packages/aws-lambda-auto-wrap/esm/README.md

```diff
@@ -1,15 +1,15 @@
-# ES Module support for AWS Lambda
+## ES Module Support for AWS Lambda
 
-The support for ES modules was added in Node v14.
-See https://aws.amazon.com/about-aws/whats-new/2022/01/aws-lambda-es-modules-top-level-await-node-js-14/
+AWS Lambda began supporting ES modules with Node v14. Learn more on the [AWS Lambda ES modules and top-level await with Node.js 14](https://aws.amazon.com/about-aws/whats-new/2022/01/aws-lambda-es-modules-top-level-await-node-js-14/) page.
 
-But AWS forgot to add support for layers. There are currently two known bugs:
+- **Current Issues (July 2024)**:
+  - Defining an ES module as a handler in a Lambda layer is not supported at present. For updates, refer to AWS support case ID 172069954500883.
 
-1. You cannot import an ES module, which is located in a layer. See https://github.com/vibe/aws-esm-modules-layer-support
-2. You cannot define an ES module as handler, which is located in a layer. See AWS support case ID 11226031451.
+- **Current Workaround**:
+  - We utilize the handler as a CommonJS module with dynamic imports.
+  - Transitioning the handler to pure ES modules does not provide immediate benefits, but we are monitoring AWS updates closely.
 
-We are affected by the second bug. When AWS fixes the underlying problem in their AWS runtime, we are able to transform the new esm handler into a real ES module.
-
-For now: We ship the ES handler as commonjs module with the help of dynamic imports.
-
-In 4.x: We remove the aws-lambda-auto-wrap npm package and manually copy over the files in the publish layer script.
\ No newline at end of file
+- **Roadmap for Improvement**:
+  - We've initiated an AWS support ticket (172069954500883) and engaged with the AWS SDK community to advocate for ES module support in Lambda layers.
+  - Future plans for version 4.x include removing the `aws-lambda-auto-wrap` npm package and enhancing our deployment process by manually integrating necessary files during the layer publishing script.
```

Comment:
```suggestion
  - In 4.x we will remove the `aws-lambda-auto-wrap` npm package, because this is an internal package. It got published by mistake. We copy the necessary files during the layer publishing script into the layer.
```